Tasks/Grunt/grunttask.js

```diff
@@ -1,22 +1,42 @@
+var fs = require('fs');
 var path = require('path');
 var tl = require('vso-task-lib');
 
-var gt = new tl.ToolRunner(tl.which('grunt', true));
+var gruntFile = tl.getPathInput('gruntFile', true);
+var cwd = tl.getInput('cwd', false);
+if (!cwd) {
+	cwd = path.dirname(gruntFile);
+}
+tl.cd(cwd);
+
+var grunt = tl.which('grunt', false);
+
+tl.debug('check path : ' + grunt);
+if(!grunt || !fs.existsSync(grunt)) {
+	var gt = new tl.ToolRunner(tl.which('node', true));	
+	var gtcli = path.resolve(cwd, 'node_modules/grunt-cli/bin/grunt');
+	
+	tl.debug('check path : ' + gtcli);
+	if(!fs.existsSync(gtcli)) {
+		tl.error('grunt-cli is neither installed globally by the user run the agent nor installed under currect working folder: ' + gtcli);
+		tl.exit(1);
+		throw ('grunt doesn\'t exist.');
```

Comment:
This prints an ugly stack trace.  process.exit(0) would be better until we get S93 static linking of the task lib